### ISSUE TITLE
Integrating a tool to debug neural networks more efficiently

### ISSUE THREAD

**Issue by FlorianDietz** [Lightning-AI/pytorch-lightning#19509]
### Description & Motivation

As an ML researcher, I find it difficult to debug neural networks efficiently when designing new architectures. There are a lot of internal states that can become anomalous, and standard inspection tools are not sufficient.

### Pitch

I recently wrote a library that can help with this, and it is already seeing use [(comgra)](https://github.com/FlorianDietz/comgra). However, it is a standalone tool and it has been suggested to me that integrating it into pytorch lightning would make it much more useful. The library I wrote extracts all relevant information from a neural network at runtime, taking care not to save too much of it and blow up your hard drive, and then visualizes it graphically in a browser. Currently, adding this library to your code requires multiple lines of code in the appropriate places. Integrating it into pytorch-lightning would make it more user friendly.

### Alternatives

The standard way to debug neural network internals is either to use tensorboard and log everything your can think of. Tensorboard is not granular enough and gets very confusing and slow when you visualize hundreds of different KPIs.

### Additional context

The library allows for a much more detailed analysis of what is happening than the usual approach of using tensorboard. You can go investigate tensors as training proceeds, drill down into individual neurons, inspect single data sets that are of special interest to you, track gradients, compare statistics between different training runs, and more.

This tool has saved me a ton of time in my research by letting me check my hypotheses much more quickly than normal and by helping me understand how the different parts of my network really interact.
